crates/router/src/middleware.rs

```diff
@@ -43,16 +43,21 @@ where
     actix_web::dev::forward_ready!(service);
 
     fn call(&self, req: actix_web::dev::ServiceRequest) -> Self::Future {
+        let old_x_request_id = req.headers().get("x-request-id").cloned();
         let mut req = req;
         let request_id_fut = req.extract::<router_env::tracing_actix_web::RequestId>();
         let response_fut = self.service.call(req);
 
         Box::pin(async move {
             let request_id = request_id_fut.await?;
+            let request_id = request_id.as_hyphenated().to_string();
+            if let Some(old_request_id) = old_x_request_id {
+                router_env::logger::info!(?request_id, ?old_request_id);
```

Comment:
should we name old request id as istio_request_id or upstream_request_id

Reply:
istio seems very specific. I would either prefer `old_request_id` or `upstream_request_id`